File: 02- February/18- Invert Binary Tree/18- Invert Binary Tree (Nayra Mohammed).py
# Author: Nayra Mohammed

# - All we need to do is traverse through the tree and for each node we reach swap it's left and right childs
  
class Solution:
    def invertTree(self, root):
        
        if not root: # no nodes here ? return null :Otherwise continue traversing 
            return root
        
        root.left,root.right= root.right,root.left #swap current node's child
        
        self.invertTree(root.left);self.invertTree(root.right) # continue traversing through these childs
     
        
        return root #return answer after being processed through all recursive calls
      
      
      
  
      
     

File: 02- February/18- Invert Binary Tree/18- Invert Binary Tree (Nayra Mohammed).cpp
//Author: Nayra Mohammed

class Solution {
public:
  
    /*
      - All we need to do is traverse through the tree and for each node we reach swap it's left and right childs
    */  
  
    TreeNode* invertTree(TreeNode* root) {
        
        if(!root) // no nodes here ? return null :Otherwise continue traversing 
            return nullptr;
      
        swap(root->left,root->right); //swap current node's child
        
        // continue traversing through these childs
        invertTree(root->right);
        invertTree(root->left);
        
        return root; //return answer after being processed through all recursive calls
    }
};
